rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -73,11 +73,16 @@ void SequentialCompressionWriter::compression_thread_fn()
             "Cannot compress message; Writer is not open!"};
   }
 
-  while (compression_is_running_) {
+  while (true) {
     std::shared_ptr<rosbag2_storage::SerializedBagMessage> message;
     std::string file;
     {
+      // This mutex synchronizes both the condition and the running_ boolean, so it has to be
+      // held when dealing with either/both
       std::unique_lock<std::mutex> lock(compressor_queue_mutex_);
+      if (!compression_is_running_) {
+        break;
+      }
       compressor_condition_.wait(lock);
```

Comment:
Because of spurious wake-ups, it is not the best practice to wait without a condition.

Reply:
Spurious wakeups are handled by this code - both queues are checked and if there is no work to do, we go to the next iteration of the while loop and wait again.